sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -220,3 +220,79 @@ def test_singular_values():
     rpca.fit(X_hat_rpca)
     assert_array_almost_equal(apca.singular_values_, [3.142, 2.718, 1.0], 14)
     assert_array_almost_equal(rpca.singular_values_, [3.142, 2.718, 1.0], 14)
+
+
+def test_svd_whiten():
+    tsvd = TruncatedSVD(n_components=X.shape[1] - 1, random_state=42,
+                        whiten=True, algorithm='arpack')
+    Xt = tsvd.fit_transform(X)
+    Xt_2 = tsvd.transform(X)
+    assert_allclose(Xt, Xt_2)
+    Xinv = tsvd.inverse_transform(Xt)
+    # Xdense is a sparse array in dense format (check zeros separately)
+    # The below equality is approximate since n_components < n_features
+    mask = Xdense == 0.0
+    assert_allclose(Xinv[mask], 0, atol=0.08)
+    assert_allclose(Xinv[~mask], Xdense[~mask], rtol=0.04)
```

Comment:
Can you move the check to the existing test, by having an `for whiten in (False, True):`?